ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -18,131 +18,214 @@
-    bloomFalsePositiveCount =
-        registry.counter(MetricRegistry.name(IndexSegment.class, name + "BloomFalsePositiveCount"));
-    keySizeMismatchCount = registry.counter(MetricRegistry.name(IndexSegment.class, name + "KeySizeMismatchCount"));
-    hardDeleteDoneCount = registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteDoneCount"));
-    hardDeleteFailedCount =
-        registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteFailedCount"));
-    hardDeleteIncompleteRecoveryCount =
-        registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteIncompleteRecoveryCount"));
-    hardDeleteExceptionsCount =
-        registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteExceptionsCount"));
-    segmentSizeForExists = registry.histogram(MetricRegistry.name(IndexSegment.class, name + "SegmentSizeForExists"));
   }
 
-  public void initializeCapacityUsedMetric(final Log log, final long capacityInBytes) {
-    currentCapacityUsed = new Gauge<Long>() {
-      @Override
-      public Long getValue() {
-        return log.getLogEndOffset();
-      }
-    };
-    registry.register(MetricRegistry.name(Log.class, name + "CurrentCapacityUsed"), currentCapacityUsed);
-    percentageUsedCapacity = new Gauge<Double>() {
-      @Override
-      public Double getValue() {
-        return ((double) log.getLogEndOffset() / capacityInBytes) * 100;
-      }
-    };
-    registry.register(MetricRegistry.name(Log.class, name + "PercentageUsedCapacity"), percentageUsedCapacity);
+  /**
+   * Create a {@link DiskLevelMetrics} object for handling metrics related to the stores on a certain disk and make it
+   * accessible for lookup.
+   * @param diskId the name of the disk.
+   * @return the {@link DiskLevelMetrics}
+   */
+  public DiskLevelMetrics createDiskLevelMetrics(String diskId) {
+    DiskLevelMetrics metrics = new DiskLevelMetrics(diskId, registry);
+    diskLevelMetrics.put(diskId, metrics);
+    return metrics;
+  }
+
+  /**
+   * @param diskId the name of the disk to lookup the metrics object for.
+   * @return the {@link DiskLevelMetrics} for a disk name, or {@code null} if that disk could not be found.
+   */
+  public DiskLevelMetrics getDiskLevelMetrics(String diskId) {
+    return diskLevelMetrics.get(diskId);
+  }
+
+  /**
+   * Metrics for a specific disk.
+   */
+  public static class DiskLevelMetrics {
+    private final Map<String, StoreLevelMetrics> storeLevelMetrics = new HashMap<>();
+    private final MetricRegistry registry;
+    public final Timer diskStartTime;
+    public final Counter diskStartFailure;
+
+    /**
+     * Create a {@link DiskLevelMetrics} object for handling metrics related to the stores on a certain disk.
+     * @param diskId the name of the disk.
+     */
+    public DiskLevelMetrics(String diskId, MetricRegistry registry) {
+      this.registry = registry;
+      String name = diskId + ".";
```

Comment:
because this name gets concatenated with the metric name. For example, name + "DiskStartTime" -> "diskN.DiskStartTime". I can change this variable name to `prefix` or something.